internal/service/db_test/handler_db_lod_test.go

```
@@ -0,0 +1,100 @@
+
+	})
+}
+
+// Test case with negative value as simplification factor
+func (t *DbTests) TestGeometrySimplificationNegativeValue() {
+	t.Test.Run("TestGeometrySimplificationNegativeValue", func(t *testing.T) {
+		path := "/collections/mock_poly/items/6?max-allowable-offset=-0.01"
+		// If lower thant minVal, then minValue (0) is considered
+		hTest.DoRequestMethodStatus(t, "GET", path, nil, nil, http.StatusOK)
+	})
+}
+
+// Test case with wrong float separator for the simplification factor
+func (t *DbTests) TestGeometrySimplificationWrongFloatSeparatorValue() {
+	t.Test.Run("TestGeometrySimplificationWrongFloatSeparatorValue", func(t *testing.T) {
+		path := "/collections/mock_poly/items?max-allowable-offset=0,01"
+		hTest.DoRequestMethodStatus(t, "GET", path, nil, nil, http.StatusBadRequest)
+	})
+}
+
+// Test case with various values as simplification factor
+func (t *DbTests) TestGeometrySimplificationVariousSimplificationValues() {
+	t.Test.Run("TestGeometrySimplificationVariousSimplificationValues", func(t *testing.T) {
+		path := "/collections/mock_poly/items/4?max-allowable-offset=0.01"
+		rr := hTest.DoRequestMethodStatus(t, "GET", path, nil, nil, http.StatusOK)
+		// Feature collection
+		var feat api.GeojsonFeatureData
+		errUnMarsh := json.Unmarshal(hTest.ReadBody(rr), &feat)
+		util.Assert(t, errUnMarsh == nil, fmt.Sprintf("%v", errUnMarsh))
+		util.Equals(t, 4, len(feat.Geom.Geometry().(orb.Polygon)[0]), "wrong number of simplified coordinates")
+
+		path = "/collections/mock_poly/items/4?max-allowable-offset=0.001"
+		rr = hTest.DoRequestMethodStatus(t, "GET", path, nil, nil, http.StatusOK)
+		errUnMarsh = json.Unmarshal(hTest.ReadBody(rr), &feat)
+		util.Assert(t, errUnMarsh == nil, fmt.Sprintf("%v", errUnMarsh))
+		util.Equals(t, 10, len(feat.Geom.Geometry().(orb.Polygon)[0]), "wrong number of simplified coordinates")
+
+		path = "/collections/mock_poly/items/4?max-allowable-offset=1"
```

May be returning an empty geometry should be considered as something not acceptable and should raise an error. To check with CrunchyData

---

@pramsey Do you have a preference?

---

may be later